doc/reverse_engineering/pathfinding.md

```diff
@@ -0,0 +1,39 @@
+# Pathfinding
+
+[Source](http://www.gamasutra.com/view/feature/131844/postmortem_ensemble_studios_age_.php?page=2)
+
+[Source](https://web.archive.org/web/20081221121604/http://www.gamasutra.com/features/gdcarchive/2000/pottinger.doc)
+
+and *Ensemble Studios Official Strategies & Secrets to Microsoft's Age of Empires II: The Age of Kings*, Bruce Shelly, 1999, published by Sybex Inc.
+
+## Algorithms
+
+Age of Empires 2 uses three different algorithms for pathfinding plus two obstruction systems.
+
+The first (high-level-)pathfinder is meant for long distances and uses mip-mapping to create a general route across the map. Criteria for the mip-map subdivision is *passibility*. It ignores moving objects, e.g. units or wildlife, though it takes "static" object like buildings into account. Whenever a building is constructed the pathfinder gets notified and updates the corresponding mip-map chains.
+
+The other two pathfinders are for short and medium distances. One uses tile-based pathfinding while the other one is a polygonal pathfinder that dynamically generates convex hulls. These are used in combat situations or when units cross paths.
```

Comment:
Well most of the things you see here is stuff confirmed by developers. I did try to rev eng more: For example, I tried to find out when exactly they switch from the high-level to the low-level pathfinder. But there are so many variables that have an influence. Formation, distance to the goal, the obstruction type, etc... Ensemble definitely put a lot of work into this. I think with a lot more time I can figure out the polygonal and tile-based pathfinders but this requires extensive testing.

I added some further reading in pathfinding.md for anyone interested (in german though).